### PR TITLE
Better API docs

### LINKED ISSUE #684: API doc for shady-render.render hides regular render docs.
Currently, the API docs for `render` show the first sentence from the `render` JSDoc, followed by the JSDoc for shady-render.render.

### PR DESCRIPTION
**Pull request by arthurevans**
Fixes some of the existing issues with API docs:

- Uses typdoc-plugin-external-module-name to consolidate multiple sub-modules into the single entry-point modules (lit-html, shady-render).

- Removes API doc for directives, which is just messed up at this point, points to template reference instead.

- Reverts to module mode, which prevents shady-dom.render from overwriting the docs for lit-html.render. (Fixes #684)

- Uses the minimal template, which places the navigation links on the left (more like the guides), creates one page per module, and seems slightly cleaner.

Staged here: https://20181212t125505-dot-polymer-lit-html.appspot.com/api/index.html

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
   "scripts": {
     "build": "tsc",
     "checksize": "rollup -c ; cat lit-html.bundled.js | gzip -9 | wc -c ; rm lit-html.bundled.js",
-    "gen-docs": "typedoc --readme ./docs/_api-readme.md --mode file --ignoreCompilerErrors --excludeNotExported --excludePrivate --out ./docs/api --gaID UA-39334307-23 ./src/lit-html.ts ./src/lib/*.ts ./src/directives/*.ts",
```
Comment (keanulee) [inline package.json]: Maybe we should move this script to docs/package.json

File: package.json [hunk 2]
```diff
@@ -44,6 +44,7 @@
     "rollup-plugin-terser": "^1.0.1",
     "tslint": "^5.11.0",
     "typedoc": "^0.13.0",
+    "typedoc-plugin-external-module-name": "^1.1.3",
```
Comment (keanulee) [inline package.json]: Consider moving these deps to docs/package.json

File: src/lit-html.ts [hunk 3]
```diff
@@ -12,6 +12,21 @@
...
+  * 
+  * -  [[html]] 
+  * -  [[svg]] 
+  * -  [[render]]
+  *   
+  * @module lit-html
+  * @preferred
+  */
+
+/** ignore this comment, which keeps typedoc from misplacing the module docs. */
```
Comment (keanulee) [inline src/lit-html.ts]: "Do not remove this comment - it keeps typedoc ..."

File: docs/_api/copy_api.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#
```
Comment (keanulee) [inline docs/_api/copy_api.sh]:
```
# This script copies the TypeScript source into a staging directory for Typedoc
# and removes the license header for typedoc-plugin-external-module-name.
```

File: package.json (hunk 1 above)
Comment (arthurevans) [reply package.json]: I encountered some problems when I tried to do so before, but I think I've figured them out. Do you know if anything else relies on this gen-docs script, or can I remove it from the top-level package.json entirely?

File: src/lit-html.ts (hunk 3 above)
Comment (arthurevans) [reply src/lit-html.ts]: 👍

File: docs/_api/copy_api.sh (hunk 4 above)
Comment (arthurevans) [reply docs/_api/copy_api.sh]: 👍

**Comment by arthurevans** [comment]
Thanks @keanulee. Will fix formatting.

I can't find the package.json changes you mentioned. Do you mean the devDependencies?

**Comment by keanulee** [comment]
package-lock, the auto-generated changes from npm. Do you have node 8 (or something not node 10)?

**Comment by keanulee** [comment]
I think you need to `git merge master` and re-run `npm run format` to make the travis-ci/pr pass.

**Pull request merged**